tests/btrfs-progs/generate_report.pm

```diff
@@ -29,23 +28,38 @@ sub log_end {
     assert_script_run($cmd);
 }
 
+# Compress all sub directories under $dir and upload them.
+sub upload_subdirs {
```

Comment:
any good reason why you packing each subfolder as separate tar and upload separately ? why not just tar whole `/opt/log/` in one shot and then upload it in one shot ?

Reply:
I prefer to one shot too. This sub just copied from xfstests scritp